File: aoc_2024/_template/1.c
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "util.h"

int processLine(char *line) {
  char **elements = splitOnChars(line, ";-,");
  int i = 0;
  while (elements[i]) {
    printf("%s;", elements[i]);
    i++;
  }
  printf("\n");
  return 1;
}

int main() {
  //init
  //read and store all lines
  char **lines = parseFile("./input.txt");
  //get number of lines
  int nbLines = 0;
  while (lines[++nbLines]);

  //actual processing
  int result = 0;
  for (int i = 0; i < nbLines; i++) {
    result += processLine(lines[i]);
  }
  printf("Result: %d\n", result);

  //de-init
  for (int i = 0; i < nbLines; i++) free(lines[i]);
  free(lines);
  return 0;
}

File: aoc_2025/day01##/1.py
lines = [l for l in open("input.txt")]

res = 0
dial = 50
for line in lines:
    if line[0] == 'L':
        dial -= int(line[1:])
        while dial < 0:
            dial += 100
    else:
        dial += int(line[1:])
        while dial >= 100:
            dial -= 100
    if dial == 0:
        res += 1

print(res)
